src/Admin/AdminInterface.php

```diff
@@ -285,12 +285,23 @@ public function setRouteBuilder(RouteBuilderInterface $routeBuilder);
     public function getRouteBuilder();
 
     /**
+     * NEXT_MAJOR: Remove this method and uncomment `subjectAsString()`.
+     *
+     * @deprecated since sonata-project/admin-bundle 3.x, use `subjectAsString()` instead.
+     *
      * @param mixed $object
      *
      * @return string
      */
     public function toString($object);
 
+    /**
+     * Returns the string representation for the admin's subject.
+     *
+     * @throws \LogicException if the admin's subject is not set
+     */
+    // public function subjectAsString(): string;
```

Comment:
Maybe introduce a new interface instead? That way the admin can implement it from day one. Or it could delegate that responsability to something else.

Reply:
I agree that these responsibilities should belong to another class.